PR연습

Park-bona opened the pull request:
PR : pull request 메인 브랜치에 병합하기 전, 팀원 및 상사에게 코드 확인 후 메인 병합 요청하는 것!

PR 생성 시 자동화 테스트 실행 되서 실패시 merge안됨. PR 습관화 하기!

Park-bona committed:
연습

youlive789 commented on src/main/java/com/dante/study/StudyApplication.java:
```diff
@@ -9,5 +9,5 @@ public class StudyApplication {
 		SpringApplication.run(StudyApplication.class, args);
 	}
-
+	//연습
```
첫 PR 멋지네융
LGTM! (Looks good to me!)

youlive789 reviewed (approved):
LGTM!

Park-bona committed:
1. h2 데이터 베이스 설치

2. DB와 Java Project 연결
	- application.properties 파일에 DB 정보 입력
	- build.gradle에 의존성 추가

3. 에러 발생
	- VS code 터미널 로그에서는 DB와 Spring Boot가 정상적으로 연결된 것 처럼 보였지만,
	  실제 연결 확인을 위해 브라우저에서 http://localhost:8080/h2-console에 접속했지만 404 에러 발생
	- Spring Boot 실행 시 코드에 /h2-console 매핑 로그가 출력되지 않음.

4. 해결을 위해 시도한 방법
	- ./gradlew clean build로 gradle 재빌드
	- H2 DB 재설치
	- 서버 연결 시 생성된 DB 파일(test.mv.db) 삭제 후 재실행

5. 에러 원인
	- Spring Boot 4.0.0-SNAPSHOT에서는 H2 콘솔 자동 설정이 누락되어 있었음.

6. 에러 해결
	- Spring Boot 3.5.3 안정 버전으로 다운그레이드 후 재빌드하니 정상적으로 브라우저에 H2 console 실행됨.

6. 회고
	- 최신 버전이어도 다른 Tool과 호환성이 떨어져 오히려 작동이 되지 않는 경우가 있을 수 있다.
	- 사용하고자 하는 라이브러리나 도구들의 버전 호환 여부를 사전에 반드시 확인해야 한다.
	- 자동 설정에 의존하지 않고, 로그 확인 하는 습관이 중요하다.
	- GUI보다 CLI를 활용하면서 로그와 설정을 직접 점검해보는 과정이 훨씬 학습에 효과적이었다

Park-bona commented:
1. h2 데이터 베이스 설치

2. DB와 Java Project 연결
	- application.properties 파일에 DB 정보 입력 
	- build.gradle에 의존성 추가

3. 에러 발생 
	- VS code 터미널 로그에서는 DB와 Spring Boot가 정상적으로 연결된 것 처럼 보였지만,
	  실제 연결 확인을 위해 브라우저에서 http://localhost:8080/h2-console에 접속했지만 404 에러 발생
	- Spring Boot 실행 시 코드에 /h2-console 매핑 로그가 출력되지 않음.

4. 해결을 위해 시도한 방법
	- ./gradlew clean build로 gradle 재빌드
	- H2 DB 재설치
	- 서버 연결 시 생성된 DB 파일(test.mv.db) 삭제 후 재실행

5. 에러 원인
	- Spring Boot 4.0.0-SNAPSHOT에서는 H2 콘솔 자동 설정이 누락되어 있었음.
 
6. 에러 해결
	- Spring Boot 3.5.3 안정 버전으로 다운그레이드 후 재빌드하니 정상적으로 브라우저에 H2 console 실행됨. 

6. 회고
	- 최신 버전이어도 다른 Tool과 호환성이 떨어져 오히려 작동이 되지 않는 경우가 있을 수 있다. 
	- 사용하고자 하는 라이브러리나 도구들의 버전 호환 여부를 사전에 반드시 확인해야 한다.
	- 자동 설정에 의존하지 않고, 로그 확인 하는 습관이 중요하다. 
	- GUI보다 CLI를 활용하면서 로그와 설정을 직접 점검해보는 과정이 훨씬 학습에 효과적이었다

7. 현재 진행 상황
	- H2에 테이블(member) 생성 및 데이터 샘플 Insert
	- memberController 생성
